src/elm/Page/Community/ActionEditor.elm

```diff
@@ -40,7 +40,7 @@ import Json.Decode as Json exposing (Value)
 import Json.Encode as Encode
 import MaskedInput.Text as MaskedDate
 import Page
-import Profile exposing (Profile)
+import Profile exposing (SelectProfile)
```

Comment:
Ended up with a cumbersome typing using extensible records. 

Thinking back maybe we can have a nested type, but it would still be weird, I loved the idea of a type that "at least have those fields", but in order to actually use it I needed to create this `Basic a` | `Minimal`, not only that extensible records don't have constructor function with the fields, so  I had to create this huge function to build the type from a list of inputs (that's how elm-graphql fills the type from the `|> with ... ` function chaining.

<img width="819" alt="Captura de Tela 2021-01-13 às 19 24 56" src="https://user-images.githubusercontent.com/1082127/104517586-07ab6480-55d5-11eb-9533-5607b0248381.png">

Reply:
I mean with nested records maybe we could have an even worst data structure to work with. Looks like Elm kinda opted out of extensible records, based on this issue: https://github.com/elm/compiler/issues/1308

So we have a half way through with Elm

Reply:
@lucca65 Yep! But we also can use a type parameter like this (based on the "Advanced Elm" course example):

```elm
type alias Profile extraInfo =
    { name : Maybe String
    , email : Maybe String
    , avatar : Avatar
    }

-- We can use `(Profile Preview)` instead of `CommunityProfile` and other short versions
type Preview
    = Preview

-- We can use `(Profile Full {...})` for the full version of the Profile
type Full
    = Full
        { bio : Maybe String
        , localization : Maybe String
        , account : Eos.Name
        , interests : List String
        , createdAt : Posix
        , communities : List CommunityInfo
        , analysisCount : Int
        , kyc : Maybe ProfileKyc
        , address : Maybe Address
        }
```

I didn't try it, but I think something like this may work fine!

Reply:
the build function is now even more complex and the usage on the app too... ): 

maybe we can leave this one like it is. It was good knowledge though!

Reply:
I feel I should explain this in more detail. Now we have to export/use more types to make this distinction. Also it is hard to match stuff too. If we want to match `account` and `communities` it becomes very verbose because now we must match `Profile` and also `Full`. Overall seems hard to use, even more hard then the regular extensible record
<img width="1484" alt="Captura de Tela 2021-01-14 às 19 22 06" src="https://user-images.githubusercontent.com/1082127/104656281-ccc13380-569d-11eb-8e80-3064fa5ca5a1.png">
<img width="1223" alt="Captura de Tela 2021-01-14 às 19 22 17" src="https://user-images.githubusercontent.com/1082127/104656296-d34fab00-569d-11eb-84dc-0b9b4f5e1b52.png">
<img width="1367" alt="Captura de Tela 2021-01-14 às 19 22 29" src="https://user-images.githubusercontent.com/1082127/104656310-db0f4f80-569d-11eb-9054-1e1a2c9078ff.png">

Reply:
also while trying to this outside of the `Profile` module, I've got some problems, maybe they are fixable, but more than 1h on this... I feel productivity slipping

Reply:
It looks like we went too far with this 😅

In my [initial comment](https://github.com/cambiatus/frontend/pull/468#discussion_r555126309) I meant just having a separate, independent type like this alongside the current `Profile` type:

```elm
type alias ProfilePreview =
    { name : Maybe String
    , account : Eos.Name
    , avatar : Avatar
    }
```

and use it instead of `CommunityProfile`, `SelectProfile`, and `ClaimProfile` because they're the same.

Reply:
I think we achieved similar results with the current structure! 💪